src/impl_/pyclass.rs

```diff
@@ -189,6 +189,204 @@ pub trait PyClassImpl: Sized {
+        #[cfg(feature = "pyproto")] async_protocol_items: &'static PyClassItems,
+        #[cfg(feature = "pyproto")] sequence_protocol_items: &'static PyClassItems,
+        #[cfg(feature = "pyproto")] buffer_protocol_items: &'static PyClassItems,
+    ) -> Self {
+        Self {
+            idx: 0,
+            pyclass_items,
+            pymethods_items,
+            #[cfg(feature = "pyproto")]
+            object_protocol_items,
+            #[cfg(feature = "pyproto")]
+            descr_protocol_items,
+            #[cfg(feature = "pyproto")]
+            gc_protocol_items,
+            #[cfg(feature = "pyproto")]
+            iter_protocol_items,
+            #[cfg(feature = "pyproto")]
+            mapping_protocol_items,
+            #[cfg(feature = "pyproto")]
+            number_protocol_items,
+            #[cfg(feature = "pyproto")]
+            async_protocol_items,
+            #[cfg(feature = "pyproto")]
+            sequence_protocol_items,
+            #[cfg(feature = "pyproto")]
+            buffer_protocol_items,
+        }
+    }
+}
+
+impl Iterator for PyClassItemsIter {
+    type Item = &'static PyClassItems;
+
+    #[cfg(not(feature = "multiple-pymethods"))]
+    fn next(&mut self) -> Option<Self::Item> {
+        match self.idx {
+            0 => {
+                self.idx += 1;
+                Some(self.pyclass_items)
+            }
+            1 => {
+                self.idx += 1;
+                Some(self.pymethods_items)
+            }
+            #[cfg(feature = "pyproto")]
+            2 => {
+                self.idx += 1;
+                Some(self.object_protocol_items)
+            }
+            #[cfg(feature = "pyproto")]
+            3 => {
+                self.idx += 1;
+                Some(self.descr_protocol_items)
+            }
+            #[cfg(feature = "pyproto")]
+            4 => {
+                self.idx += 1;
+                Some(self.gc_protocol_items)
+            }
+            #[cfg(feature = "pyproto")]
+            5 => {
+                self.idx += 1;
+                Some(self.iter_protocol_items)
+            }
+            #[cfg(feature = "pyproto")]
+            6 => {
+                self.idx += 1;
+                Some(self.mapping_protocol_items)
+            }
+            #[cfg(feature = "pyproto")]
+            7 => {
+                self.idx += 1;
+                Some(self.number_protocol_items)
+            }
+            #[cfg(feature = "pyproto")]
+            8 => {
+                self.idx += 1;
+                Some(self.async_protocol_items)
+            }
+            #[cfg(feature = "pyproto")]
+            9 => {
+                self.idx += 1;
+                Some(self.sequence_protocol_items)
+            }
+            #[cfg(feature = "pyproto")]
+            10 => {
+                self.idx += 1;
+                Some(self.buffer_protocol_items)
+            }
+            _ => None,
```

Comment:
I find it tricky that this case handles `2..=10` or not depending on the `pyproto` feature. I cannot suggest a nicer way, but maybe a comment would be useful?

Reply:
Yes, I agree that this is a bit horrible (esp. with the second lot of duplication for `multiple-pymethods`). In the end I decided not to bother refactoring further as (hopefully) 0.17 series will be the last set of releases with `pyproto`.

Reply:
Pushed a comment as suggested.